EIPS/eip-5114.md

```diff
@@ -0,0 +1,75 @@
+---
+eip: 5114
+title: Soulbound Token
+description: A token that is attached to a "soul" at mint time and cannot be transferred after that.
+author: Micah Zoltu (@MicahZoltu)
+discussions-to: https://ethereum-magicians.org/t/eip-5114-soulbound-token/9417
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-05-30
+requires: 721
+---
+
+## Abstract
+A soulbound token is a token that is bound to another Non-Fungible Token (NFT; e.g., a EIP-721 token) when it is minted, and cannot be transferred/moved after that.
+
+## Specification
+```solidity
+interface IERC5114 {
+	// fired anytime a new instance of this token is minted
+	// this event **MUST NOT** be fired twice for the same `tokenId`
+	event Mint(uint256 indexed tokenId, address indexed nftAddress, uint256 indexed nftTokenId);
+
+	// returns the NFT token that owns this token.
+	// this function **MUST** throw if the token hasn't been minted yet
+	// this function **MUST** always return the same result every time it is called after it has been minted
+	// this function **MUST** return the same value as found in the original `Mint` event for the token
+	function ownerOf(uint256 index) external view returns (address nftAddress, uint256 nftTokenId);
+	
+	// returns a censorship resistant URI with details about this token collection
+	// the metadata returned by this is merged with the metadata return by `tokenUri(uint256)`
+	// the tokenUri **MUST** be immutable and content addressable (e.g., ipfs://)
+	// the tokenUri **MUST NOT** point at mutable/censorable content (e.g., https://)
+	// data from `tokenUri` takes precedence over data returned by this method
+	// any external links referenced by the content at `tokenUri` also **MUST** follow all of the above rules
+	function collectionUri() external view returns (string tokenUri);
```

Comment:
Should all of these `tokenUri` references (and return variable name) be `collectionUri`? Seems a bit odd to return a `tokenUri` from the `collectionUri` function.

Reply:
```suggestion
	// returns a censorship resistant URI with details about this token collection
	// the metadata returned by this is merged with the metadata return by `tokenUri(uint256)`
	// the collectionUri **MUST** be immutable and content addressable (e.g., ipfs://)
	// the collectionUri **MUST NOT** point at mutable/censorable content (e.g., https://)
	// data from `tokenUri` takes precedence over data returned by this method
	// any external links referenced by the content at `collectionUri` also **MUST** follow all of the above rules
	function collectionUri() external view returns (string collectionUri);
```